### ISSUE TITLE
apt_repository is incorrectly setting the file permissions on new repositories

### ISSUE THREAD

**Issue by jamiejamison** [ansible/ansible-modules-core#4064]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

apt_repository
##### ANSIBLE VERSION

```
ansible 2.1.0.0
```
##### OS / ENVIRONMENT

Ubuntu 14.04 (Trusty Tahr)
##### SUMMARY

When using the apt_repository module the permissions of the repository listings in /etc/apt/sources.list.d are set to mode 0420 (-r---w----) unless the mode option is specified and the value for the mode is quoted.
##### STEPS TO REPRODUCE

```
- name: Add the Adiscon v8-stable PPA to the list of repositories
  apt_repository:
    repo: "ppa:adiscon/v8-stable"
    state: present
    update_cache: yes

- name: Add the Alestic repository
  apt_repository:
    repo: "deb http://ppa.launchpad.net/alestic/ppa/ubuntu {{ ansible_distribution_release }} main"
    state: present
    update_cache: yes
    mode: 0644
```

##### EXPECTED RESULTS

I expected the files

/etc/apt/sources.list.d/ppa_adiscon_v8_stable_trusty.list
/etc/apt/sources.list.d/ppa_launchpad_net_alestic_ppa_ubuntu.list

to be created with mode 0644.
##### ACTUAL RESULTS

The files 

/etc/apt/sources.list.d/ppa_adiscon_v8_stable_trusty.list
/etc/apt/sources.list.d/ppa_launchpad_net_alestic_ppa_ubuntu.list

were created with mode 0420.

```
TASK [configure_rsyslog : Add the Adiscon v8-stable PPA to the list of repositories] ***
changed: [******] => {"changed": true, "diff": {}, "invocation": {"module_args": {"filename": null, "install_python_apt": true, "mode": "420", "repo": "ppa:adiscon/v8-stable", "state": "present", "update_cache": true, "validate_certs": true}, "module_name": "apt_repository"}, "repo": "ppa:adiscon/v8-stable", "state": "present"}

TASK [ec2_consistent_snapshot : Add the Alestic repository] ********************
changed: [sec-app-2.prod.doxo.com] => {"changed": true, "diff": {}, "invocation": {"module_args": {"filename": null, "install_python_apt": true, "mode": "420", "repo": "deb http://ppa.launchpad.net/alestic/ppa/ubuntu trusty main", "state": "present", "update_cache": true, "validate_certs": true}, "module_name": "apt_repository"}, "repo": "deb http://ppa.launchpad.net/alestic/ppa/ubuntu trusty main", "state": "present"}
```

This crept in somewhere between 2.0.2.0 and 2.1.0.0. The documentation for apt_repository states that the default file mode for creating new files is 0644, which is decimal 420. Somewhere along the line the octal value 0644 is being converted to decimal and that decimal value is being passed to the call to os.chmod which is interpreting it as an octal value when setting the permissions.

**Comment by jctanner** [comment]
related to https://github.com/ansible/ansible-modules-core/pull/3991

**Comment by abadger** [comment]
This should be fixed in 2.1.1rc3 via cherry-pick to stable-2.1  da177b1
